azurerm/resource_arm_databricks_workspace.go

```diff
@@ -0,0 +1,244 @@
+		Timeout:    60 * time.Minute,
+		MinTimeout: 15 * time.Second,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for Databricks Workspace (%s) to become available: %s", name, err)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmDatabricksWorkspaceRead(d, meta)
+}
+
+func resourceArmDatabricksWorkspaceUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).databricksWorkspacesClient
+	ctx := meta.(*ArmClient).StopContext
+	log.Printf("[INFO] preparing arguments for Azure ARM Databricks Workspace update.")
+
+	name := d.Get("workspace_name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	tags := d.Get("tags").(map[string]interface{})
+	expandedTags := expandTags(tags)
+
+	workspaceUpdate := databricks.WorkspaceUpdate{
+		Tags: expandedTags,
+	}
+
+	_, err := client.Update(ctx, workspaceUpdate, resGroup, name)
+	if err != nil {
+		return err
+	}
+
+	read, err := client.Get(ctx, resGroup, name)
+	if err != nil {
+		return err
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Databricks Workspace %s (resource group %s) ID", name, resGroup)
+	}
+
+	log.Printf("[DEBUG] Waiting for Databricks Workspace (%s) to become available", name)
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"Updating", "Creating"},
+		Target:     []string{"Succeeded"},
+		Refresh:    workspaceStateRefreshFunc(ctx, client, resGroup, name),
+		Timeout:    60 * time.Minute,
+		MinTimeout: 15 * time.Second,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for Databricks Workspace (%s) to become available: %s", name, err)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmDatabricksWorkspaceRead(d, meta)
+}
+
+func resourceArmDatabricksWorkspaceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).databricksWorkspacesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["workspaces"]
+
+	resp, err := client.Get(ctx, resGroup, name)
+
+	if err != nil {
+		return fmt.Errorf("Error Retrieving Databricks Workspace %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	// covers if the resource has been deleted outside of TF, but is still in the state
+	if utils.ResponseWasNotFound(resp.Response) {
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("Error making Read request on Azure Databricks Workspace %s: %s", name, err)
+	}
+
+	d.Set("workspace_name", name)
+	d.Set("resource_group_name", resGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	if sku := resp.Sku; sku != nil {
+		d.Set("sku", sku.Name)
+	}
+
+	if props := resp.WorkspaceProperties; props != nil {
+		d.Set("managed_resource_group_id", props.ManagedResourceGroupID)
```

Comment:
at the moment this'll fail since there's no field specified in the schema for it - but this'll be resolved based on the comment above